contracts/AgentMech.sol

```diff
@@ -69,16 +97,14 @@ contract AgentMech is ERC721Mech {
     mapping(uint256 => uint256[2]) public mapRequestIds;
     // Map of request Id => sender address
     mapping(uint256 => address) public mapRequestAddresses;
-    // Map of account nonces
-    mapping(address => uint256) public mapNonces;
 
     /// @dev AgentMech constructor.
     /// @param _token Address of the token contract.
     /// @param _tokenId The token ID.
     /// @param _price The minimum required price.
-    constructor(address _token, uint256 _tokenId, uint256 _price) ERC721Mech(_token, _tokenId) {
-        // Check for the token address
-        if (_token == address(0)) {
+    constructor(address _mechMarketplace, address _token, uint256 _tokenId, uint256 _price) ERC721Mech(_token, _tokenId) {
```

Comment:
Each agent mech registers the marketplace (can be changed by the operator / owner)